napari_aicssegmentation/widgets/collapsible_box.py

```diff
@@ -54,7 +55,7 @@ def _create_content_box(self):
         content_box = QFrame()
         content_box_layout = QVBoxLayout()
         content_box_layout.setContentsMargins(9, 9, 9, 9)
-        content_box_layout.addLayout(self._content)
+        content_box_layout.addLayout(self._content_layout)
```

Comment:
Nested layout does work, however I believed I've learned that it's not the ideal way to deal with dynamic content. I think the safe / more QT friendly way to do this would be to override the `QWidget.addWidget` method instead and have the added widgets be added to the collapsible box's layout. 
I think it would look like this:
~~~python
def addWidget(widget: QWidget):
   self._content_box_layout().addWidget(widget)
~~~ 
The idea being that you just add widgets to the box without needing to worry about layouts. If you then want to add a group of widgets positioned within a layout you can either group your widgets inside a `QFrame` and add that frame to the box.
Alternatively if you really want to work directly with layouts, you could either:

1. Expose the content layout with `get_content_layout()` method or property
2. override the `layout()` method to return the content's layout instead of the whole widget's layout

So a few alternatives to consider. I just feel that forcing to pass in a Layout object at creation is a bit restrictive and in general not a best practice. I won't die on this hill though :D

Reply:
Thanks. I tried making a getter and setter for `_content_layout` but I ran into some issues -- namely, the way the code is structured right now, if I were to instantiate a `CollapsibleBox` and then set the `content_layout`, then I also have to make the object re-create the content box and re-add it to the overall layout. I suggest we can file this refactoring as an issue and address it separately from this PR.

Reply:
Sounds good. Can you just add a TODO comment in the code so we remember?